src/JustSaying/Messaging/Middleware/DefaultSqsMiddleware.cs

```diff
@@ -0,0 +1,63 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Amazon.SQS.Model;
+using JustSaying.Messaging.Channels;
+using Microsoft.Extensions.Logging;
+
+namespace JustSaying.Messaging.Middleware
+{
+    public class DefaultSqsMiddleware : MiddlewareBase<GetMessagesContext, IList<Amazon.SQS.Model.Message>>
+    {
+        private readonly ILogger<DefaultSqsMiddleware> _logger;
+
+        public DefaultSqsMiddleware(ILogger<DefaultSqsMiddleware> logger) : base(null)
+        {
+            _logger = logger;
+        }
+
+        protected override async Task<IList<Message>> RunInnerAsync(GetMessagesContext context, Func<Task<IList<Message>>> func)
+        {
+            try
+            {
+                var results = await func().ConfigureAwait(false);
+
+                _logger.LogTrace(
+                    "Polled for messages on queue '{QueueName}' in region '{Region}', and received {MessageCount} messages.",
+                    context.QueueName,
+                    context.RegionName,
+                    context.Count);
+
+                return results;
+            }
+            catch (InvalidOperationException ex)
+            {
+                _logger.LogTrace(
+                    ex,
+                    "Could not determine number of messages to read from queue '{QueueName}' in '{Region}'.",
+                    context.QueueName,
+                    context.RegionName);
+            }
+            catch (OperationCanceledException ex)
+            {
+                _logger.LogTrace(
+                    ex,
+                    "Suspected no message on queue '{QueueName}' in region '{Region}'.",
```

Comment:
Maybe put something about a timeout in the message to give a clue to why this suspicion?

Reply:
Updated the message to better explain what's happening.